BoardView.cpp

```diff
@@ -320,22 +326,87 @@ void BoardView::HandleInput() {
 		float mwheel = io.MouseWheel;
 		if (mwheel != 0.0f) {
 			const ImVec2 &target = io.MousePos;
-			ImVec2 coord = ScreenToCoord(target.x, target.y);
 			mwheel *= 0.5f;
 			// Ctrl slows down the zoom speed:
 			if (ImGui::IsKeyDown(17)) {
 				mwheel *= 0.1f;
 			}
-			m_scale = m_scale * powf(2.0f, mwheel);
 
-			ImVec2 dtarget = CoordToScreen(coord.x, coord.y);
-			ImVec2 td = ScreenToCoord(target.x - dtarget.x, target.y - dtarget.y, 0);
-			m_dx += td.x;
-			m_dy += td.y;
-			m_needsRedraw = true;
+			float factor = powf(2.0f, mwheel);
+			ChangeZoom(target, factor);
 		}
 	}
 	if (!io.WantCaptureKeyboard) {
+
+		//Ctrl-Q as alternative quit shortcut (Alt-F4 is already handled by Windows)
+		if (ImGui::IsKeyPressed('Q') && ImGui::IsKeyDown(17)) {
+			m_wantsQuit = true;
+		}
```

Comment:
Could you have `Ctrl+O` and `Ctrl+Q` together? Would probably mean moving this to the top of  `Update` method.